### PR TITLE
Audioplayer repeats

### PR DESCRIPTION
**Pull request by mmahalwy**

### CODE REVIEW

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:32769

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:32780

File: src/components/Audioplayer/Segments/index.js [hunk 1]
```diff
@@ -1,290 +1,56 @@
 import React, { Component, PropTypes } from 'react';
-import ReactDOM from 'react-dom';
-import { decrypt } from 'sjcl';
+import Helmet from 'react-helmet';
+
+import debug from 'helpers/debug';
```
Comment (sabeurthabti) [inline src/components/Audioplayer/Segments/index.js]: How is this alias configured?

File: src/components/Audioplayer/index.js [hunk 2]
```diff
@@ -1,286 +1,312 @@
...
 
 // Helpers
 import debug from '../../helpers/debug';
-import scroller from '../../scripts/utils/scroller';
+import scroller from '../../utils/scroller';
 
 const style = require('./style.scss');
 
 @connect(
-  state => ({
```
Comment (sabeurthabti) [inline src/components/Audioplayer/index.js]: refactor to remove decorators, see container folder for example.

**Comment by ahmedre** [comment]
Deployed to: http://staging.quran.com:32794

File: src/components/Audioplayer/index.js [hunk 3]
```diff
@@ -26,34 +17,7 @@ import scroller from '../../utils/scroller';
...
-    next,
-    previous,
-    setRepeat,
-    toggleScroll,
-    buildOnClient,
-    update
-  }
-)
-export default class Audioplayer extends Component {
+export class Audioplayer extends Component {
```
Comment (sabeurthabti) [inline src/components/Audioplayer/index.js]: nice! exporting the class to make it testable.

File: src/components/Audioplayer/index.js [hunk 4]
```diff
@@ -475,3 +451,21 @@ export default class Audioplayer extends Component {
     );
   }
 }
+
+const mapStateToProps = (state, ownProps) => ({
```
Comment (sabeurthabti) [inline src/components/Audioplayer/index.js]: :ok_hand:

File: src/components/Audioplayer/RepeatDropdown/index.js [hunk 5]
```diff
@@ -98,21 +100,37 @@ export default class RepeatButton extends Component {
...
+    const { repeat, setRepeat, surah } = this.props;
+    const array = Array(surah.ayat).join().split(',');
 
     return (
-      <Col md={12}>
-        <h5>
-          Repeating Ayah: {current}
-        </h5>
+      <Col md={12} style={{paddingTop: 15}}>
+        Ayah: <br />
```
Comment (sabeurthabti) [inline src/components/Audioplayer/RepeatDropdown/index.js]: place Ayah in a `<p>` tag

File: src/components/Audioplayer/Segments/index.js (hunk 1 above)
Comment (mmahalwy) [inline src/components/Audioplayer/Segments/index.js]: @sabeurthabti not an alias but prefilling the namespace for the debug lib. https://github.com/quran/quran.com-frontend/blob/master/src/helpers/debug.js

**Comment by sabeurthabti** [comment]
UI reviewed and code reviewed. All Good.

**Comment by mmahalwy** [comment]
thank you sir! LEts merge this

**Pull request merged**